### PR TITLE
Enable parametric w3c/b3/datadog/b3 single trace context tests for Go

### PR DESCRIPTION
**Pull request by dianashevchenko**
## Description

Updating golang code in parametric tests to be able to run b3multi/datadog/w3c propagator tests. Most of the changes remove skipping of the tests or change it to a 'bug' format

## Check list

Your PR is not ready to be reviewed? Please save it as a draft :pray:

- [ ] Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
- [ ] CI is passing
- [ ] There is at least one approval from code owners

Yes to all? Feel free to merge it whenever you want :heart:

### CODE REVIEW

File: parametric/test_headers_b3multi.py [hunk 1]
```diff
@@ -53,7 +52,10 @@ def test_headers_b3multi_extract_valid(test_agent, test_library):
 
 
 @enable_b3multi()
-@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library(
+    "golang",
+    "BUG: the parent_id will be 0 in any span that didn't have a parent context ; if extract fails - the parent context is nil",
```
Comment (ajgajg1134) [inline parametric/test_headers_b3multi.py:57]: I ran into this exact same issue in the b3 tests. I believe the parent_id being 0 is the correct behavior and perhaps the test should change?
I would recommend:
```
assert span.get("span_id") != 0
assert span.get("parent_id") == 0
```

Comment (dianashevchenko) [reply parametric/test_headers_b3multi.py:57]: The context extracted with this propagator will be later used in `tracer.ChildOf(ctx)` option, so the `assert span.get("span_id") != 0` is technically incorrect. As spanID of the context IS the parentID of the span in question. 
Maybe it's possible to introduce an exception for Go though

File: parametric/test_headers_b3multi.py [hunk 2]
```diff
@@ -53,7 +52,7 @@ def test_headers_b3multi_extract_valid(test_agent, test_library):
 
 
 @enable_b3multi()
-@pytest.mark.skip_library("golang", "not implemented")
+@pytest.mark.skip_library("golang", "BUG: the parent_id will be 0 in any span that didn't have a parent context ; if extract fails - the parent context is nil")
```
Comment (katiehockman) [inline parametric/test_headers_b3multi.py]: Let's link to the GH issue for this so we can track it. Here and throughout. If it's something that's intended behavior that we don't intend to fix, then let's make sure to indicate that clearly.

**Review by katiehockman — APPROVED** [review]
Looks good, but I think there might have been an issue with a merge? There's a ton of commits in here before your changes (starting with `92d48b916d9d6967cef10f54810ef26f2cbd2e07`) that probably shouldn't be in your PR. Maybe need to do a rebase?

File: parametric/test_headers_b3multi.py (hunk 2 above)
Comment (dianashevchenko) [reply parametric/test_headers_b3multi.py]: This was mentioned in a different pr for node, and will be fixed later

File: parametric/test_headers_b3.py [hunk 3]
```diff
@@ -57,8 +55,8 @@ def test_headers_b3_extract_invalid(test_agent, test_library):
 
 @enable_b3()
 @pytest.mark.skip_library("dotnet", "Latest release does not implement new configuration")
-@pytest.mark.skip_library("golang", "not impemented")
-@pytest.mark.skip_library("nodejs", "not impemented")
+@pytest.mark.skip_library("nodejs", "not implemented")
```
Comment (Kyle-Verhoog) [inline parametric/test_headers_b3.py:58]:
```suggestion
```

File: parametric/test_headers_datadog.py [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
+def enable_datadog_propagator() -> Any:
+    env = {
+        "DD_TRACE_PROPAGATION_STYLE_EXTRACT": "datadog",
+        "DD_TRACE_PROPAGATION_STYLE_INJECT": "datadog",
+    }
+    return parametrize("library_env", [env])
+
 
-@pytest.mark.skip_library("golang", "not implemented")
+@enable_datadog_propagator()
```
Comment (Kyle-Verhoog) [inline parametric/test_headers_datadog.py:21]: Is this change made because the default isn't `datadog` for go? If not, can we revert this change as it would be introducing an unnecessary change.

(also the logic of the tests would be slightly different, we'd be testing the explicit datadog behaviour rather than the default of each library)

**Review by Kyle-Verhoog — COMMENTED** [review]
just a couple small things otherwise looks good!

File: parametric/test_headers_b3.py (hunk 3 above)
Comment (dianashevchenko) [reply parametric/test_headers_b3.py:58]: I was just fixing a typo but okay 😅

File: parametric/test_headers_datadog.py (hunk 4 above)
Comment (dianashevchenko) [reply parametric/test_headers_datadog.py:21]: The default now it 'tracecontext,Datadog'. But let me double-check

File: parametric/test_headers_b3.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply parametric/test_headers_b3.py:58]: oh wait, I think this was my bad lol. For some reason it looked like we added another `nodejs` entry. All good for the typo fix lol

**Review by Kyle-Verhoog — APPROVED** [review]
lgtm! maybe worth getting @zacharycmontoya's review here too 😄

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

**Comment by cbeauchesne** [comment]
@dianashevchenko parametric test is failing on golang ?

**Review by cbeauchesne — APPROVED** [review]
